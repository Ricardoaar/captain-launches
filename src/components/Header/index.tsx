import React from "react";
import { Link, useLocation } from "react-router-dom";

const Header: React.FC = () => {
  const location = useLocation();
  const path = location.pathname.split("/");
  const endpoint = path[path.length - 1];
  return (
    <header className="header bg-secondary text-font text-center">
      <nav>
        <ul className="grid  grid-cols-2 items-center py-4 px-1 sm:px-4">
          <li className={endpoint === "" ? "link-lighted" : "link"}>
            <Link to="/">Home</Link>
          </li>
          <li className={endpoint === "upcoming" ? "link-lighted" : "link"}>
            <Link to="/launches/upcoming">Upcoming</Link>
          </li>
        </ul>
      </nav>
    </header>
  );
};

export default Header;
