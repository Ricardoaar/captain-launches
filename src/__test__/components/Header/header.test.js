import React from "react";
import { mount, shallow } from "enzyme";
import Header from "../../../components/Header";
import { create } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe("<Header />", () => {
  test("Rendering header", () => {
    const wrapper = shallow(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(wrapper.length).toBe(1);
  });

  test("Rendering the title", () => {
    const wrapper = mount(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(wrapper.find("li").first().text()).toEqual("Home");
  });

  test("should match the snapshot", () => {
    const wrapper = create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
