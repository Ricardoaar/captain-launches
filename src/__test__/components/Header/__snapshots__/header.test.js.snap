// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match the snapshot 1`] = `
<header
  className="header bg-secondary text-font text-center"
>
  <nav>
    <ul
      className="grid  grid-cols-2 items-center py-4 px-1 sm:px-4"
    >
      <li
        className="link-lighted"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="link"
      >
        <a
          href="/launches/upcoming"
          onClick={[Function]}
        >
          Upcoming
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
